org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/migration/IDoValueMigrationHandler.java

```diff
@@ -0,0 +1,151 @@
+ * exact point in time, when the structure migration was introduced.
+ * <p>
+ * To avoid non-idempotent value migrations, do not re-use renamed or deleted ID values in the (future) code base. If
+ * non-idempotent value migrations cannot be avoided, consider using an {@link IDoStructureMigrationHandler} instead (if
+ * you control the namespace of the relevant data objects and are able to introduce a new type version).
+ * <p>
+ * <b>Example 1:</b> built-in ID FOO is merged / replaced by built-in ID BAR<br>
+ * Note: this migration is only idempotent, if value FOO is not used anymore in the future code base.
+ *
+ * <pre>
+ * public class FooDoValueMigrationHandler extends AbstractDoValueMigrationHandler<FooTypeId> {
+ *
+ *   public static final DoValueMigrationId ID = DoValueMigrationId.of("3740d4ce-849b-449e-afb1-0986993c9463");
+ *
+ *   &#64;Override
+ *   public DoValueMigrationId id() {
+ *     return ID;
+ *   }
+ *
+ *   &#64;Override
+ *   public Class<? extends ITypeVersion> typeVersion() {
+ *     return Foo_2.class;
+ *   }
+ *
+ *   &#64;Override
+ *   public FooTypeId migrate(DoStructureMigrationContext ctx, FooTypeId value) {
+ *     return "foo".equals(value.unwrap()) ? FooTypes.BAR : value;
+ *   }
+ * }
+ * </pre>
+ *
+ * <b>Example 2:</b> Match a DO entity data object and migrate a specific (string) attribute. Make sure to return a
+ * modified copy of the input value instead of manipulating the provided input value itself, such that the caller is
+ * able to detect the change by an equals-comparison.
+ *
+ * <pre>
+ * public class BarDoValueMigrationHandler extends AbstractDoValueMigrationHandler<BarDo> {
+ *
+ *   public static final DoValueMigrationId ID = DoValueMigrationId.of("f73240b8-ca5d-4e4a-a257-9d4deb067b68");
+ *
+ *   protected static final String PREFIX = "prefix-";
+ *
+ *   &#64;Override
+ *   public DoValueMigrationId id() {
+ *     return ID;
+ *   }
+ *
+ *   &#64;Override
+ *   public Class<? extends ITypeVersion> typeVersion() {
+ *     return Bar_3.class;
+ *   }
+ *
+ *   &#64;Override
+ *   public BarDo migrate(DoStructureMigrationContext ctx, BarDo value) {
+ *     if (value.getName().startsWith(PREFIX)) {
+ *       return value; // no migration required
+ *     }
+ *
+ *     return BEANS.get(DataObjectHelper.class).clone(value) // clone provided value to allow change detection by caller
+ *         .withName(PREFIX + value.getName());
+ *   }
+ * }
+ * </pre>
+ */
+@ApplicationScoped
+public interface IDoValueMigrationHandler<T> {
+
+  /**
+   * Unique identifier for this value migration handler.
+   */
+  DoValueMigrationId id();
+
+  /**
+   * @return Class of value node that needs to be migrated (assignable check is performed so ExampleId will also be
+   *         migrated in an IId node).
+   */
+  Class<T> valueClass();
+
+  /**
+   * Type version used for sorting.
+   * <p>
+   * The type version defined on a value migration is not related in any way to the type versions of the migrated data
```

Comment:
Rephrase. They are not releated in a way that there are executed in between the structure migrations, but still related so that the same type versions are used for ordering.
Remove first sentence, use the second sentence as new first one. Then add "The value migrations are applied after all structure migrations, thus the type version doesn't directly related to a structure migration handler with the same type version."